### PR TITLE
feat: add default intention

### PR DESCRIPTION
**Pull request by AoyuQC**
1. add default when no intention index added

### CODE REVIEW

**Review by NingLu — APPROVED** [review]

File: source/lambda/online/lambda_intention_detection/intention.py [hunk 1]
```diff
@@ -21,15 +21,34 @@ def lambda_handler(state:dict, context=None):
...
-        "kwargs": doc['answer']['jsonlAnswer'].get('kwargs', {}),
-        } for doc in res['result']['docs'] if doc['score'] > 0.4
-    ]
+
+    if not res['result']['docs']:
+        # add default intention
+        import json
+        import pathlib
+        current_path = pathlib.Path(__file__).parent.resolve()
+        with open(f'{current_path}/intention_utils/default_intent.jsonl', 'r') as json_file:
```
Comment (yike5460) [inline source/lambda/online/lambda_intention_detection/intention.py:30]: try:
    with open(f'{current_path}/intention_utils/default_intent.jsonl', 'r') as json_file:
        json_list = list(json_file)
except FileNotFoundError:
    print(f"File not found: {current_path}/intention_utils/default_intent.jsonl")
    json_list = []
except json.JSONDecodeError as e:
    print(f"Error decoding JSON: {e}")
    json_list = []

File: source/lambda/online/lambda_intention_detection/intention.py [hunk 2]
```diff
@@ -21,15 +21,34 @@ def lambda_handler(state:dict, context=None):
...
-        "score": doc['score'],
-        "name": doc['answer']['jsonlAnswer']['intent'],
-        "intent": doc['answer']['jsonlAnswer']['intent'],
-        "kwargs": doc['answer']['jsonlAnswer'].get('kwargs', {}),
-        } for doc in res['result']['docs'] if doc['score'] > 0.4
-    ]
+
+    if not res['result']['docs']:
+        # add default intention
+        import json
```
Comment (yike5460) [inline source/lambda/online/lambda_intention_detection/intention.py:27]: move to top of the file

File: source/lambda/online/lambda_intention_detection/intention.py [hunk 3]
```diff
@@ -21,15 +21,34 @@ def lambda_handler(state:dict, context=None):
...
+
+        for json_str in json_list:
+            intent_result = json.loads(json_str)
+            intention_fewshot_examples = [{
+                "query": intent_result["question"],
+                "score": 'n/a',
+                "name": intent_result['answer']['intent'],
+                "intent": intent_result['answer']['intent'],
+                "kwargs": intent_result['answer'].get('kwargs', {}),
+            }]
```
Comment (yike5460) [inline source/lambda/online/lambda_intention_detection/intention.py:41]: except KeyError as e:
        print(f"Missing key in JSON data: {e}")
    except json.JSONDecodeError as e:
        print(f"Error decoding JSON string: {e}")

File: source/lambda/online/lambda_main/main.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ def lambda_handler(event_body:dict, context:dict):
     custom_message_id = event_body.get("custom_message_id", "")
     user_id = event_body.get("user_id", "default_user_id")
     # TODO Need to modify key
-    group_name = event_body.get("chatbot_config").get("default_workspace_config",{"rag_workspace_ids":["Admin"]}).get("rag_workspace_ids")[0]
+    group_name = event_body.get("chatbot_config").get("default_workspace_config",{"rag_workspace_ids":["Admin"]}).get("rag_workspace_ids","Admin")[0]
```
Comment (yike5460) [inline source/lambda/online/lambda_main/main.py:82]: why admin?

**Review by yike5460 — COMMENTED** [review]
comments leaved

File: source/lambda/online/lambda_main/main.py (hunk 4 above)
Comment (AoyuQC) [reply source/lambda/online/lambda_main/main.py:82]: this is default index created when deployed

**Pull request merged**